core/tx_list.go

```diff
@@ -344,18 +344,18 @@ func (l *txList) Add(tx *types.Transaction, priceBump uint64) (bool, *types.Tran
 		}
 		// thresholdFeeCap = oldFC  * (100 + priceBump) / 100
 		a := big.NewInt(100 + int64(priceBump))
-		aFeeCap := new(big.Int).Mul(a, oldGasFeeCap)
-		aTip := a.Mul(a, oldGasTipCap)
+		thresholdFeeCap := new(big.Int).Mul(a, oldGasFeeCap)
+		thresholdTip := a.Mul(a, oldGasTipCap)
 
 		// thresholdTip    = oldTip * (100 + priceBump) / 100
 		b := big.NewInt(100)
-		thresholdFeeCap := aFeeCap.Div(aFeeCap, b)
-		thresholdTip := aTip.Div(aTip, b)
+		newFeeCap := new(big.Int).Mul(b, newGasFeeCap)
+		newTip := a.Mul(b, newGasTipCap)
```

Comment:
The comments don't match the code anymore.

Reply:
Nice catch, let's agree on the solution before I fix those.

Reply:
Is the old code really problematic? The threshold is rounded down, so replacing is somewhat easier than with the fully accurate calculation, right?

Reply:
Looks like neither of the two issues (rounding errors on wei-level fee values and overflow) could realistically happen in real life, so not dividing by 100 looks okay but the old code doesn't looks outrageously wrong either.

Reply:
Agreed, both look fine. To avoid unnecessary change, I would stay with the old version unless some need for the change can be shown.

Reply:
> Is the old code really problematic? The threshold is rounded down, so replacing is somewhat easier than with the fully accurate calculation, right?

Exactly, without this change it is possible to replace transactions without increasing the gas price. In Ethereum this case is handled by [this early return](https://github.com/celo-org/celo-blockchain/blob/e91fdf12b79379259e5bee0d56e687af86ad98a5/core/tx_list.go#L320). It also breaks [this assumption in the ported test](https://github.com/celo-org/celo-blockchain/blob/e91fdf12b79379259e5bee0d56e687af86ad98a5/core/tx_pool_test.go#L2620-L2623), that's why I started thinking about it in the first place.

Reply:
We could increase the numbers in the test to more realistic levels to make it pass or use your suggested solution. No strong opinions on this.

Reply:
If we decide that we don't want to change the behavior I'd just remove that assertion from the test.

Reply:
I removed the changes and adapted the test to more realistic values.